cluster-autoscaler/utils/kubernetes/factory.go

```diff
@@ -0,0 +1,18 @@
+package kubernetes
+
+import (
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+	kube_client "k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	kube_record "k8s.io/kubernetes/pkg/client/record"
+	v1core "k8s.io/kubernetes/pkg/client/clientset_generated/clientset/typed/core/v1"
+
+	"github.com/golang/glog"
+)
+
+// CreateEventRecorder creates an event recorder to send custom events to Kubernetes to be recorded for targeted Kubernetes objects
+func CreateEventRecorder(kubeClient kube_client.Interface) kube_record.EventRecorder {
+	eventBroadcaster := kube_record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: kubeClient.Core().Events("")})
```

Comment:
Thanks! Turns out that it is caused by the fact I'm blindly creating various podListers and nodeListers like:

```yaml
func NewStaticAutoscaler(opts AutoscalingOptions, predicaterChecker *simulator.PredicateChecker, kubeClient kube_client.Interface, kubeEventRecorder kube_record.EventRecorder) *StaticAutoscaler {
	autoscalingContext := NewAutoscalingContext(opts, predicaterChecker, kubeClient, kubeEventRecorder)

	scaleDown := NewScaleDown(autoscalingContext)
	stopchannel := make(chan struct{})
	unschedulablePodLister := kube_util.NewUnschedulablePodLister(kubeClient, stopchannel)
	scheduledPodLister := kube_util.NewScheduledPodLister(kubeClient, stopchannel)
	readyNodeLister := kube_util.NewReadyNodeLister(kubeClient, stopchannel)
	allNodeLister := kube_util.NewAllNodeLister(kubeClient, stopchannel)
```

which fails when kubeClient is a fake one.

Almost certainly I'd need a way to inject those dependencies via a set of parameters possible wrapped inside a struct.
Do you want me to do that in this PR?